static/app/components/core/compactSelect/gridList/option.tsx

```diff
@@ -105,7 +102,9 @@ export function GridListOption({node, listState, size}: GridListOptionProps) {
             )}
           </CheckWrap>
         )}
-        {leadingItems}
+        {typeof leadingItems === 'function'
+          ? leadingItems({disabled: isDisabled, isFocused, isSelected})
+          : leadingItems}
```

Comment:
this is an adjacent fix where we couldn’t pass `leadingItems` as a function, but it worked for `trailingItems`